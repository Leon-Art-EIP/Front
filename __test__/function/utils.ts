import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
