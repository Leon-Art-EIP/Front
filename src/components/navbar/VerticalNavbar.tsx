import { useState } from "react";
import IconButton from "../single-art-page/artwork/IconButton";
import { INavbarProps } from "./Navbar";
import { Menu } from "@mui/icons-material";

export default function VerticalNavbar({ link: Link, ...props }: Omit<INavbarProps, "image">): JSX.Element {
  const [open, setOpen] = useState(false);

  return (
    <div className="flex flex-col w-full h-full">
      <div className="flex justify-start items-center h-24 bg-white w-full border-b border-b-secondaryGrey">
        <IconButton
          backgroundColor="white"
          color="black"
          icon={Menu}
          onClick={() => {
            setOpen(!open);
          }}
        />
      </div>
      {open && (
        <div className="absolute z-10 flex bg-white mt-[95px] rounded-b-xl">
          <div className="flex flex-col justify-center items-center gap-5 py-2">
            {props.tabs.map(
              (tab) =>
                tab.verticalNavbar && (
                  <Link
                    key={`${tab.name}-${tab.href}`}
                    href={`${tab.href == "/profile" ? `/profile/${props.userId}` : tab.href}`}
                  >
                    <div style={{ color: `${props.selectedTabHref === tab.href ? "red" : "black"}` }} className="px-2">
                      {tab.name}
                    </div>
                  </Link>
                )
            )}
          </div>
        </div>
      )}
    </div>
  );
}
