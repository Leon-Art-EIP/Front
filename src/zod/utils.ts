export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
