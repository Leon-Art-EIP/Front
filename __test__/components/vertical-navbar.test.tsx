import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import VerticalNavbar from "../../src/components/navbar/VerticalNavbar";

const mockLink = ({ children, href }: { children: JSX.Element; href: string }) => <a href={href}>{children}</a>;

const mockTabs = [
  { name: "Home", href: "/home", loggedIn: true, navbar: true, verticalNavbar: true, header: false },
  { name: "Profile", href: "/profile", loggedIn: true, navbar: false, verticalNavbar: true, header: false },
  { name: "Settings", href: "/settings", loggedIn: true, navbar: false, verticalNavbar: true, header: true },
];

test("VerticalNavbar snapshot with open menu", () => {
  testSnapshot(<VerticalNavbar tabs={mockTabs} selectedTabHref="/profile" link={mockLink} userId="123" />);
});

test("VerticalNavbar snapshot with closed menu", () => {
  testSnapshot(<VerticalNavbar tabs={mockTabs} selectedTabHref="/home" link={mockLink} userId="456" />);
});

test("VerticalNavbar snapshot with no tabs", () => {
  testSnapshot(<VerticalNavbar tabs={[]} selectedTabHref="/" link={mockLink} userId="789" />);
});
