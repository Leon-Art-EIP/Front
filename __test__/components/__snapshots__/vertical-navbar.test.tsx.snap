// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VerticalNavbar snapshot with closed menu 1`] = `
<div
  className="flex flex-col w-full h-full"
>
  <div
    className="flex justify-start items-center h-24 bg-white w-full border-b border-b-secondaryGrey"
  >
    <button
      className="rounded-full white flex gap-4 px-6 py-2.5"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        style={
          {
            "color": "black",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`VerticalNavbar snapshot with no tabs 1`] = `
<div
  className="flex flex-col w-full h-full"
>
  <div
    className="flex justify-start items-center h-24 bg-white w-full border-b border-b-secondaryGrey"
  >
    <button
      className="rounded-full white flex gap-4 px-6 py-2.5"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        style={
          {
            "color": "black",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`VerticalNavbar snapshot with open menu 1`] = `
<div
  className="flex flex-col w-full h-full"
>
  <div
    className="flex justify-start items-center h-24 bg-white w-full border-b border-b-secondaryGrey"
  >
    <button
      className="rounded-full white flex gap-4 px-6 py-2.5"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        style={
          {
            "color": "black",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
  </div>
</div>
`;
